src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsauditapi/listeners/HMPPSAuditListener.kt

```diff
@@ -20,15 +27,22 @@ class HMPPSAuditListener(
   @JmsListener(destination = "\${sqs.queue.name}")
   fun onAuditEvent(message: String) {
     log.debug("Received message $message")
-    val auditEvent: AuditEvent = gson.fromJson(message, AuditEvent::class.java)
+    val auditEvent: AuditEvent = mapper.readValue(message, AuditEvent::class.java)
     auditService.audit(auditEvent)
   }
 
+  @Entity
+  @Table(name = "AuditEvent")
   data class AuditEvent(
+    @Id
+    @GeneratedValue
+    val id: UUID? = null,
```

Comment:
Ooh I was under the impression that data classes were a no-no for JPA.  Is this working? :pray:

Reply:
Ah!  I guess as there will be no updates the data class is fine?

Reply:
It all seems ok from the tests (phew)

Reply:
Since the AuditEvent is immutable, we never need to read and amend

Reply:
I use data classes for JPA in court register